service/interpreter/cadence/workflowProvider.go

```diff
@@ -254,3 +269,30 @@ func (w *workflowProvider) GetUnhandledSignalNames(ctx interpreter.UnifiedContex
 	}
 	return workflow.GetUnhandledSignalNames(wfCtx)
 }
+
+func (s *selector) Select(ctx interpreter.UnifiedContext) {
+	wfCtx, ok := ctx.GetContext().(workflow.Context)
+	if !ok {
+		panic("cannot convert to cadence workflow context")
+	}
+
+	s.selector.Select(wfCtx)
+}
+
+func (s *selector) ReceiveSignalValueBlocking(ctx interpreter.UnifiedContext, signalName string) interface{} {
```

Comment:
I believe we don't need to use selector. 

See how the ReceiveAsync is implemented:
```
func (t *cadenceReceiveChannel) ReceiveAsync(valuePtr interface{}) (ok bool) {
	return t.channel.ReceiveAsync(valuePtr)
}
```
It should be similar code